src/components/pill/pill.spec.tsx

```diff
@@ -589,6 +612,85 @@ describe("Pill", () => {
               });
             });
           });
+
+          describe("when pillRole is status and isDarkbackground prop is set to true", () => {
+            const pillRole = "status";
+            const styleSet = styleConfig(true)[pillRole];
+
+            describe.each(darkStyleTypes)(
+              'when the pill style is set as "%s"',
+              (style) => {
+                if (style !== "neutralWhite") {
```

Comment:
comment: avoid conditional describe blocks

This becomes a lot more confusing by having a conditional statement wrapping the describes. I'd prefer to see us separate out the tests a little if this is needed. This is where nested describes become a little cumbersome in my personal opinion and we've previously relied on using too many `.each()` blocks which makes the test files less verbose and harder to read/scan